<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="styles.css">
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
        integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
    <title>REPORT</title>
</head>

<body>
    <h4>16S rRNA REPORT</h4>
    <h5 style="color: black;">CORE-METRICS SUMMARY</h5>

    
        <div class="row">
            <div class="col-md-6">
                <img src="images/alpha-compare (1).svg" alt="evenness">
                <a href="images/alpha-compare (1).svg" target="_blank">Link to image</a>

                <p>Alpha diversity evenness distribution</p>
               
            </div>
            <div class="col-md-6">
                <img src="images/alpha-compare (2).svg" alt="shannon">
                <a href="images/alpha-compare (2).svg"target="_blank">Link to image</a>
                <div class="bottom-left">Shannon matrix alpha diversity</div>
            </div>
        </div>
        <div class="row">
    
            <div class="col-md-6">
                <img src="images/emperor (7).png" alt="jaccard">
                <a href="images/emperor (7).png"target="_blank">Link to image</a>
                <div class="bottom-left">Jaccard matrix beta diversity, scatter for Dogs</div>
              
            </div>
            <div class="col-md-6">
                <img src="images/emperor (8).png" alt="jaccard">
                <a href="images/emperor (8).png"target="_blank">Link to image</a>
                <div class="bottom-left">Jaccard matrix beta diversity, scatter sample ID</div>
                
            </div>
        </div>
        <div class="row">
            <div class="col-md-6">
                <img src="images/emperor (6).png" alt="bray curtis">
                <a href="images/emperor (6).png"target="_blank">Link to image</a>
                <div class="bottom-left">Bray Curtis matrix beta diversity scatter against Dog</div>
               
            </div>
            <div class="col-md-6">
                <img src="images/emperor (5).png" alt="bray curtis">
                <a href="images/emperor (5).png"target="_blank">Link to image</a>
                <div class="bottom-left">Bray Curtis matrix beta diversity scatter against sample ID</div>
               
            </div>
        </div>
 

        <p style="text-align:center; margin-top: 12px;"><a href="images/tree.svg" target="_blank">Phylogeny Tree with OTUS</a></p>
  
    <div id="quiz1">
        <h4>INPUT DATA ASSESSMENT</h4>
        <p> 1.Were the number, length and quality of the reads obtained in line with what would be expected for the
            sequencing platform used?</p>
        <P>Yes - </P>
        <p>2.Was the input dataset of sufficiently good quality to perform the analysis?
            Yes- Because all the sequences had a pass in the sequence quality also, no much work was done in
            pre-proccessing
        </p>
        <p>3.How did the readsâ€™ quality and GC content affect the way analysis was run?
        </p>
    </div>
    <div id="quiz2">
        <h4>OPERATIONAL ASSESSMENT</h4>
        <ol>
            <li>Was the choice affected by the nature and/or quality of the reads?</li>
            <li>Was the choice made due to the time and cost of the analysis?</li>
            <li>What are the accuracy and performance considerations for the chosen piece of software?
            </li>
           
        </ol>


    </div>
    <footer>
        <div class="crit">
            <h5>Critique</h5>
            <ol>
                <li>PEAR is slightly slower</li>
                <li>UCHIME is more accurate in chimera detection than Chimeraslayer</li>
                <li>QIIME2 requires a manifest file which you have to put all sample ids, their absolute paths and direction of
                    sequences which could be taxing</li>
                <li>pyNAST requires python version 2.7-2.8 to install</li>
                <li>Mothur is on galaxy but requires fast internet.</li>
                <p style="text-align:center;"><a
                        href="https://training.galaxyproject.org/archive/2019-02-07/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html">Tutorial
                        for Mothur on galaxy</a></p>
            </ol>

        </div>
        <h4>Conclusions</h4>
        <ul>
            <li>Uparse had an accurate algorithmn for pairing that PEAR(96%)</li>
            <li>41% of chimeras detected</li>
            <li>Quality scores for both raw reaads and trimmed reads were high inferring that sequencing plattform was good and primers </li>
           
        </ul>
    </footer>
</body>

</html>