<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
        integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
    <link rel="stylesheet" href="styles.css">
    <!-- JQuery -->
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <!-- Bootstrap tooltips -->
    <script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.4/umd/popper.min.js"></script>
    <!-- Bootstrap core JavaScript -->
    <script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.5.0/js/bootstrap.min.js"></script>
    <!-- MDB core JavaScript -->
    <script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.19.1/js/mdb.min.js"></script>
    <!-- Font Awesome -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css">
    <!-- Google Fonts -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap">
    <!-- Bootstrap core CSS -->
    <link href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.5.0/css/bootstrap.min.css" rel="stylesheet">
    <!-- Material Design Bootstrap -->
    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.19.1/css/mdb.min.css" rel="stylesheet">


    <title>REPORT</title>
</head>

<body>
    <h4>16S rRNA WORKFLOW REPORT</h4>

    <div class="intro">
        <h5>INTRODUCTION</h5>
        <p> The 16S rRNA is a ribosomal RNA necessary for the synthesis of all prokaryotic proteins. The genes
            coding for the 16S rRna has regions that are highly conserved and are shared
            in all bacteria species making it useful in identifying bacteria in samples. Moreover,
            the gene has variable regions which can be used to reconstruct phylogenies.

        </p>
    </div>

    <div>
        <p>In this miniproject we created two worklows that can be used to determine microbial communities present in
            different enivironment using the 16S rRNA analysis.</p>
        <p>Here is a summary flow chart of tools used on the workflows using three phases namely:</p>
        <ol>
            <li>Preprocessing Reads.</li>
            <li>OTU picking,Classification and Phylogenetic Tree Generation.</li>
            <li>Measure Diversity and other Statistical Analyses.</li>
        </ol>
    </div>
    <div id="quiz1">
        <h4>Input Data Assessment</h4>
        <ul>
            <li>The input datasets were of good quality based on the FASTQ results</li>
            <div class="row">
                <div class="col-md-6">
                    <img src="images/fastq_before.png" alt="">
                    <p style="text-align: center;">Before Trimmomatic</p>

                </div>
                <div class="col-md-6">
                    <img src="images/trimmomaticsfastq.png" alt="">
                    <p style="text-align: center;">After Trimmomatic</p>

                </div>
            </div>
            <p>How did the GC content affect analysis?</p>

        </ul>
    </div>

    <div class="row">
        <div class="col-md-6">
            <h4>QIIME Workflow</h4>
            <img src="images/E2E QIIME.png" style=" border: 5px solid #555; height: 500px;" alt="qiime workflow">
            <h5>Runtime Analysis</h5>
            <ul>
                <li>Time Taken: 106mins</li>

                <li>Hardware Usage- Fair ,user can perform other light tasks while running.</li>
                <li>Storage - 8.9 G.B </li>
            </ul>
        </div>
        <div class="col-md-6">
            <h4>Multiple-Tools Workflow</h4>
            <img src="images/uchime.png" style=" border: 5px solid #555; height: 500px;" alt="uparse/uchime">
            <h5>Runtime Analysis</h5>
            <ul>
                <li>Time Taken - 2.5 hrs.</li>
                <li>Hardware Usage - Intensive,computer freezes.</li>
                <li>Storage - 8.8 G.B</li>
            </ul>


        </div>
    </div>
    <div id="quiz2">
        <h4>Operational Assessment</h4>
        <h5>Factors considered </h5>
        <ul>
            <li>Speed</li>
            <li>Accuracy</li>
            <li>Cost</li>
            <li>Output Formats and Reports</li>
            <li>Hardware Requirements</li>
        </ul>
    </div>

    <h5>Alternative Tools Critique</h5>
    <ol>
        <li>Trimmomatic vs Prinseq: No difference in fastqc results. </li>
        <li>UPARSE (96% merged) vs PEAR(99% merged): UPARSE does not require input of reverse reads unlike PEAR. PEAR
            produces 4 outputs.</li>
        <li>UCHIME vs Chimeraslayer: Uchime is faster and detects more chimeras than Chimeraslayer which also requires
            different formats.</li>
        <li>UPARSE vs QIIME : Qiime picked 209 OTUS while UPARSE had 190 OTUS.</li>

        <li>UCLUST vs QIIME: UCLUST uses less time than QIIME.</li>
        <li>QIIME vs Mothur : Mothur deletes all reads in the preparation stage whereas QIIME picked 209 OTUS using
            deblur plugin.</li>
        <a
            href="https://training.galaxyproject.org/archive/2019-02-07/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html">Mothur
            Tutorial</a>
        <li>MAFFT vs pyNAST for Alignment: Couldnt download pyNAST cause it requires python version 2.7 and below.</li>
    </ol>

    <h5>Results Analysis/Metrics Comparions</h5>
    <p>This is a comparison of the alpha and beta diversity analyses from the two workflows.</p>


    <div class="row">
        <div class="col-md-6">
            <img src="images/alpha-compare (3).svg" alt="qiime shannon">
            <p> Alpha Diversity Shannon Index(q.w)</p>
            <a href="images/alpha-compare (3).svg" target="_blank">Link to image</a>



        </div>
        <div class="col-md-6">
            <img src="images/alpha-compare (2).svg" alt="shannon">
            <div class="bottom-left">Alpha Diversity Shannon Index(o.w)</div>
            <a href="images/alpha-compare (2).svg" target="_blank">Link to image</a>

        </div>
    </div>
    <div class="row">

        <div class="col-md-6">
            <img src="images/emperor (11).png" alt="jaccard">
            <div class="bottom-left">Bray Curtis Emperor(q.w)</div>
            <a href="images/emperor (11).png" target="_blank">Link to image</a>


        </div>
        <div class="col-md-6">
            <img src="images/emperor (10).png" alt="jaccard">
            <div class="bottom-left">Bray Curtis Emperor(o.w)</div>
            <a href="images/emperor (10).png" target="_blank">Link to image</a>

        </div>
    </div>



    <div class="container my-4">
        <p>Other Images from Qiime workflow....</p>

        <a class="btn btn-primary" href="images/level-7-bars (1).svg">Taxonomy Barplot</a>
        <a class="btn btn-indigo" href="images/alpha-rarefaction.png">Alpha rarefaction/Shannon</a>
        <a class="btn btn-indigo" href="images/rareobserved.png">Alpha rarefaction/Observed otus</a>
        <a class="btn btn-dark-green" href="images/deblur-stats.png">Deblur Stats</a>
        <a class="btn btn-purple" href="images/tree (1).svg">Rooted Tree</a>
        <a class="btn btn-primary" href="images/tree (2).svg">Unrooted Tree</a>
        <a class="btn btn-primary" href="images/bVc7ka1Dm3yUi2DczboC1g.svg">Circular Tree</a>
        <a class="btn btn-primary" href="images/metadata (2).tsv">Taxonomy Table</a>

        <a class="btn btn-dark-green" href="images/unweighted_unifrac.png">Unifrac Unweighted</a>

        <a class="btn btn-indigo" href="images/emperor (12).png">Unweighted unifrac/Emperor</a>

    </div>


    <footer>
        <h4>Summary</h4>
        <ul>
            <li>Quality Filtered -Good.</li>
            <li>Trimmed-Not many reads were trimmed(roughly 24-40).</li>
            <li>99% stitched with pear.</li>
            <li>43% Chimeras detected.</li>
            <li>SILVA Database vs Greengeens : Silva is wide and more curated than greengeens.</li>
            <li>190 and 209 OTUS picked.</li>
            <li>Rarefaction sequence depth- G is most dense followed by K then B.But the curve flattened.</li>
            <li>OTU correlated with any treatments or metadata - B is the most diverse(Least effective treatment cause has diverse species) followed by K
                then G.</li>
                <p style="font-weight: bolder;">G is the most effective treatment followed by K then B is the least effective.</p>
        </ul>
        <div id="conc">
            <h4>Conclusion</h4>
            <p>1.QIIME end to end is better - all factors considered.</p>
            <p>2.The other workflow is lighter and inclusive.</p>

        </div>

        <h3 style="text-align: center;">THE END!!!</h3>
    </footer>
</body>

</html>